autoload/ale/events.vim

```diff
@@ -11,8 +11,32 @@ function! ale#events#QuitRecently(buffer) abort
     return l:time && ale#util#ClockMilliseconds() - l:time < 1000
 endfunction
 
+function! ale#events#EnterEvent(buffer) abort
+    " When entering a buffer, we are no longer quitting it.
+    call setbufvar(a:buffer, 'ale_quitting', 0)
+
+    " Save the original filetype for later comparison.
+    let l:filetype = getbufvar(a:buffer, '&filetype')
+    call setbufvar(a:buffer, 'ale_original_filetype', l:filetype)
+
+    " Don't run costly, pointless operations on startup.
+    if v:vim_did_enter
```

Comment:
One thing you'll have to check is, do you get results for linting a file if you open it from the command line? Say if you run `vim whatever.py`, does it still run `pylint` and give you results right away, even if the option for linting when the filetype changes is off?